src/observersTools/kinematicsTools.cpp

```diff
@@ -0,0 +1,190 @@
+    case false: // the velocity is expressed in the local frame
+      kine.linVel = kine.orientation.toMatrix3() * vel.linear();
+      kine.angVel = kine.orientation.toMatrix3() * vel.angular();
+  }
+  switch(accIsGlobal)
+  {
+    case true: // the velocity is expressed in the global frame
+      kine.linAcc = acc.linear();
+      kine.angAcc = acc.angular();
+      break;
+    case false: // the velocity is expressed in the local frame
+      kine.linAcc = kine.orientation.toMatrix3() * acc.linear();
+      kine.angAcc = kine.orientation.toMatrix3() * acc.angular();
+  }
+
+  return kine;
+}
+
+void addToLogger(const stateObservation::kine::Kinematics & kine, mc_rtc::Logger & logger, const std::string & prefix)
+{
+  logger.addLogEntry(prefix + "_position",
+                     [&kine]() -> const stateObservation::Vector3
+                     {
+                       if(kine.position.isSet()) { return kine.position(); }
+                       else { return stateObservation::Vector3::Zero(); }
+                     });
+  logger.addLogEntry(prefix + "_ori",
+                     [&kine]() -> Eigen::Quaterniond
+                     {
+                       if(kine.orientation.isSet()) { return kine.orientation.inverse().toQuaternion(); }
+                       else { return stateObservation::kine::Orientation::zeroRotation().toQuaternion(); }
+                     });
+  logger.addLogEntry(prefix + "_linVel",
+                     [&kine]() -> const stateObservation::Vector3
+                     {
+                       if(kine.linVel.isSet()) { return kine.linVel(); }
+                       else { return stateObservation::Vector3::Zero(); };
+                     });
+  logger.addLogEntry(prefix + "_angVel",
+                     [&kine]() -> const stateObservation::Vector3
+                     {
+                       if(kine.angVel.isSet()) { return kine.angVel(); }
+                       else { return stateObservation::Vector3::Zero(); };
+                     });
+  logger.addLogEntry(prefix + "_linAcc",
+                     [&kine]() -> const stateObservation::Vector3
+                     {
+                       if(kine.linAcc.isSet()) { return kine.linAcc(); }
+                       else { return stateObservation::Vector3::Zero(); };
+                     });
+  logger.addLogEntry(prefix + "_angAcc",
+                     [&kine]() -> const stateObservation::Vector3
+                     {
+                       if(kine.angAcc.isSet()) { return kine.angAcc(); }
+                       else { return stateObservation::Vector3::Zero(); };
+                     });
+}
+
+void removeFromLogger(mc_rtc::Logger & logger, const std::string & prefix)
+{
+  logger.removeLogEntry(prefix + "_position");
+  logger.removeLogEntry(prefix + "_ori");
+  logger.removeLogEntry(prefix + "_linVel");
+  logger.removeLogEntry(prefix + "_angVel");
+  logger.removeLogEntry(prefix + "_linAcc");
+  logger.removeLogEntry(prefix + "_angAcc");
+}
```

Comment:
This way this can be simplified as:
```suggestion
void removeFromLogger(const stateObservation::kine::Kinematics & kine, mc_rtc::Logger & logger)
{
  logger.removeLogEntries(&kine);
}
```